src/jit/morph.cpp

```diff
@@ -979,25 +954,25 @@ fgArgInfo::fgArgInfo(GenTreeCall* newCall, GenTreeCall* oldCall)
     else
     {
         assert(oldCall->gtCallObjp != nullptr);
-        newArgObjp.Current() = newCall->gtCallArgs;
-        newArgs              = &newArgObjp;
-        oldArgObjp.Current() = oldCall->gtCallObjp;
-        oldArgs              = &oldArgObjp;
+        newArgObjp.SetNode(newCall->gtCallObjp);
```

Comment:
Old code was broken, it was setting `Current` to `gtCallArgs` instead of `gtCallObjp` which made no sense. I don't think this code is ever hit, it's only used when calls are cloned and that's not exactly common. The only case I know about is loop hoisting but only a few helper calls can be considered invariant and hoisted. And those helper calls do not have `this`...

Speaking of `this`, I find it extremely odd that it's stored separately from rest of the args, it `gtCallObjp`. It complicates things and as seen here this can lead to potential bugs.